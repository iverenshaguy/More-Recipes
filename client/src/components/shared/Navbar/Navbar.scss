@import '../../../assets/scss/modules';

.navbar-brand {
  color: $white !important;
}

#more-recipes-navbar {
  background-color: $transparent;
  padding: 1rem;
  // transition: all 0.3s ease-out;
  transition: 0.3s ease-out;
  width: 100%;

  #navbar-right {
    float: right;
  }

  .affix {
    background-color: $white;
    box-shadow: $navbar-affix-shadow;

    .nav-item > a,
    .navbar-brand,
    .fa-heart {
      color: $darkgrey !important;
    }

    &.navbar {
      a {
        &.nav-link {
          color: $darkgrey !important;
        }
      }
    }
  }
}

.input-group {
  &.blur {
    z-index: -1 !important;
  }
}

.search-icon {
  a {
    padding: 0.46rem 0.5rem 0 0;

    &:hover,
    &:focus,
    &:active {
      color: $green !important;
      cursor: pointer;
      text-decoration: none;
    }
  }
}

@media only screen and (min-width: 1200px) {
  #more-recipes-navbar {
    #search-form {
      width: 70%;
    }
  }
}

@media only screen and (min-width: 992px) and (max-width: 1199px) {
  #more-recipes-navbar {
    #search-form {
      width: 60%;
    }
  }
}

@media only screen and (min-width: 768px) and (max-width: 991px) {
  #more-recipes-navbar {
    #search-form {
      width: 50%;
    }
  }
}

@media only screen and (min-width: 576px) and (max-width: 767px) {
  #more-recipes-navbar {
    #search-form {
      display: none !important;
    }

    .search-icon {
      a {
        color: $white !important;
        // padding: 0.5rem 1rem 0 0 ;
      }
    }
  }
}

@media only screen and (max-width: 575px) {
  #more-recipes-navbar {
    #search-form {
      display: none;
    }
  }
}

.navbar {
  a {
    &.nav-link {
      color: $white !important;
    }
  }

  .dropdown-menu {
    position: absolute !important;
    right: 0% !important;
  }

  .btn {
    background-color: transparent;
    border: 1px solid $white;
    color: $white;
    outline: none;

    &:hover,
    &:focus,
    &:active {
      background-color: $green;
      border: 1px solid $green;
      color: $white;
    }
  }

  .fa-heart {
    color: $white;
  }

  li {
    &.active {
      a {
        //color: #light-green;
        border-bottom: 2px solid $white !important;
      }
    }
  }
}

// Navbaar on scroll

.affix {
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1030;

  #search {
    border: 1px solid $grey !important;
    border-right: 0 !important;
    box-shadow: none !important;
  }

  button {
    background-color: transparent;
    border: 1px solid $grey !important;
    color: $navbar-btn !important;
    outline: none;
  }

  .search-btn {
    border: 1px solid $grey !important;
    border-left: 0 !important;
    box-shadow: none !important;
    color: $light-green !important;
  }

  &.navbar {
    li {
      &.active {
        a {
          // color: $light-green;
          border-bottom: 2px solid $darkgrey !important;
        }
      }
    }
  }
}

.nav-profile-picture-div {
  border-radius: 50%;
  height: 2rem;
  text-transform: capitalize;
  width: 2rem;
  // margin-top: -0.06rem;

  img {
    border-radius: 50%;
    height: 2rem;
    margin-top: -0.3rem;
    text-transform: capitalize;
    width: 2rem;
  }

  .dropdown-menu {
    width: 20%;
  }
}

.notifications-link {
  .dropdown-menu {
    left: 58% !important;
    width: 40% !important;
  }
}
