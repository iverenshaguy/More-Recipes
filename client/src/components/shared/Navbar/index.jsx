import React from 'react';
import {
  Nav,
  Navbar,
  DropdownMenu,
  DropdownToggle,
  UncontrolledDropdown,
} from 'reactstrap';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import { userPropTypes } from '../../../helpers/proptypes';
import './Navbar.scss';

/**
 * Returns NavbarWrapper Component
 * @export
 * @function NavbarWrapper
 * @param {object} props
 * @returns {component} NavbarWrapper
 */
const NavbarWrapper = (props) => {
  const { user, isAuthenticated } = props;
  const altImage = 'images/user-image-placeholder.png';

  if (!isAuthenticated) {
    return (
      <div>
        <Navbar color="faded" light expand="md" id="more-recipes-navbar">
          <Link to="/" className="navbar-brand">
            MORE RECIPES
          </Link>
        </Navbar>
      </div>
    );
  }

  return (
    <Navbar color="faded" light className="navbar-expand" id="more-recipes-navbar">
      <Link to="/" className="navbar-brand">
        MORE RECIPES
      </Link>
      <div className="collapse navbar-collapse justify-content-between" id="navbarNav">
        <Nav className="ml-auto" navbar>
          <UncontrolledDropdown nav className="justify-content-start notifications-link">
            <DropdownToggle nav>
              <FontAwesome name="bell" size="lg" />
            </DropdownToggle>
            <DropdownMenu right>
              <div className="container-fluid">
                <p>You have a new review</p>
                <div className="dropdown-divider" />
                <p>3 people viewed your recipe</p>
                <div className="dropdown-divider" />
                <p>Review your security details</p>
                <div className="dropdown-divider" />
                <p>Chef Steve just posted a new recipe</p>
                <div className="dropdown-divider" />
                <p>1 person added your recipe to their favorites</p>
              </div>
            </DropdownMenu>
          </UncontrolledDropdown>
          <UncontrolledDropdown nav>
            <div className="nav-profile-picture-div rounded-cirle menu-dropdown d-inline">
              <DropdownToggle nav>
                <img
                  src={user && user.profilePic ? user.profilePic : altImage}
                  onError={(e) => { e.target.src = altImage; }}
                  className="rounded-cirle"
                  alt="User"
                />
              </DropdownToggle>
              <DropdownMenu right>
                <Link to={`/${user && user.username}`} className="dropdown-item">
                  Profile
                </Link>
                <Link to={`/${user && user.username}/profile/edit`} className="dropdown-item">Edit Profile</Link>
                {/* <div className="dropdown-divider" /> */}
                {/* <a className="dropdown-item" id="my-recipes" href="user-profile.html">
                    My Recipes
                  </a> */}
                {/* <div className="dropdown-divider" /> */}
                {/* <a className="dropdown-item">Settings</a> */}
                <div className="dropdown-divider" />
                <a className="dropdown-item" href="/login" onClick={props.logout}>
                  Log out
                </a>
              </DropdownMenu>
            </div>
          </UncontrolledDropdown>
          <UncontrolledDropdown nav>
            <DropdownToggle nav className="menu-dropdown">
              <FontAwesome name="reorder" size="lg" />
            </DropdownToggle>
            <DropdownMenu right>
              <Link
                to="/recipes/new"
                className="dropdown-item add-recipe"
                href="/recipes/new"
                title="New Recipe"
              >
                Add a Recipe
              </Link>
              {/* <a className="dropdown-item get-favorites" href="user-profile.html">
                  Favorite Recipes
                </a> */}
              <Link to="/" className="dropdown-item">
                Top Recipes
              </Link>
              <a
                className="dropdown-item"
                href="#favorite-recipes"
                title="favorite-recipes"
              >
                Favorite Recipes
              </a>
            </DropdownMenu>
          </UncontrolledDropdown>
        </Nav>
      </div>
    </Navbar>
  );
};

NavbarWrapper.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired,
  showAddRecipeModal: PropTypes.func.isRequired,
  ...userPropTypes
};

export default NavbarWrapper;
