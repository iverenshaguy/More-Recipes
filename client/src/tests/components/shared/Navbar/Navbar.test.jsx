import React from 'react';
import Navbar from '../../../../components/shared/Navbar';

const logoutMock = jest.fn();

const authProps = {
  isAuthenticated: false,
  type: 'auth',
  showAddRecipeModal: jest.fn(),
  user: { firstname: 'Dave', lastname: 'Smith' },
  logout: logoutMock
};

const normalProps = {
  isAuthenticated: true,
  type: 'home',
  showAddRecipeModal: jest.fn(),
  user: { firstname: 'Dave', lastname: 'Smith' },
  logout: logoutMock
};

const profileImageProps = {
  ...normalProps,
  user: {
    ...normalProps.user,
    profilePic: 'pic'
  }
};

describe('Navbar', () => {
  it('renders correctly when app is not authenticated', () => {
    const wrapper = shallow(<Navbar {...authProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when type = home', () => {
    const wrapper = shallow(<Navbar {...normalProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('img[src="images/user-image-placeholder.png"]').length).toEqual(1);
  });

  it('renders user-image correctly when not provided', () => {
    const wrapper = shallow(<Navbar {...profileImageProps} />);
    expect(wrapper.find('img[src="pic"]').length).toEqual(1);
  });

  it('logs user out on logout click', () => {
    const wrapper = shallow(<Navbar {...normalProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('a[href="/login"]').simulate('click', { preventDefault() { } });

    expect(logoutMock).toHaveBeenCalled();
  });
});
