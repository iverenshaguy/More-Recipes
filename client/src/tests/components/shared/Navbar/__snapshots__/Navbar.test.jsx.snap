// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar logs user out on logout click 1`] = `
<Navbar
  className="navbar-expand"
  color="faded"
  expand={false}
  id="more-recipes-navbar"
  light={true}
  tag="nav"
>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    MORE RECIPES
  </Link>
  <div
    className="collapse navbar-collapse justify-content-between"
    id="navbarNav"
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <UncontrolledDropdown
        className="justify-content-start notifications-link"
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          color="secondary"
          nav={true}
        >
          <FontAwesome
            name="bell"
            size="lg"
          />
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <div
            className="container-fluid"
          >
            <p>
              You have a new review
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              3 people viewed your recipe
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              Review your security details
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              Chef Steve just posted a new recipe
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              1 person added your recipe to their favorites
            </p>
          </div>
        </DropdownMenu>
      </UncontrolledDropdown>
      <UncontrolledDropdown
        nav={true}
      >
        <div
          className="nav-profile-picture-div rounded-cirle menu-dropdown d-inline"
        >
          <DropdownToggle
            aria-haspopup={true}
            color="secondary"
            nav={true}
          >
            <img
              alt="User"
              className="rounded-cirle"
              src="images/user1.jpg"
            />
             
            Dave
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <Link
              className="dropdown-item"
              replace={false}
              to="/user/profile"
            >
              Profile
            </Link>
            <Link
              className="dropdown-item"
              replace={false}
              to="/user/profile/edit"
            >
              Edit Profile
            </Link>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/login"
              onClick={[Function]}
            >
              Log out
            </a>
          </DropdownMenu>
        </div>
      </UncontrolledDropdown>
      <UncontrolledDropdown
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          className="menu-dropdown"
          color="secondary"
          nav={true}
        >
          <FontAwesome
            name="reorder"
            size="lg"
          />
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <a
            className="dropdown-item add-recipe"
            data-toggle="modal"
            href="#add-edit-modal"
            title="New Recipe"
          >
            Add a Recipe
          </a>
          <Link
            className="dropdown-item"
            replace={false}
            to="/recipes"
          >
            All Recipes
          </Link>
          <Link
            className="dropdown-item"
            replace={false}
            to="/"
          >
            Top Recipes
          </Link>
        </DropdownMenu>
      </UncontrolledDropdown>
    </Nav>
  </div>
</Navbar>
`;

exports[`Navbar renders correctly when app is not authenticated 1`] = `
<div>
  <Navbar
    color="faded"
    expand="md"
    id="more-recipes-navbar"
    light={true}
    tag="nav"
  >
    <Link
      className="navbar-brand"
      replace={false}
      to="/"
    >
      MORE RECIPES
    </Link>
  </Navbar>
</div>
`;

exports[`Navbar renders correctly when type = home 1`] = `
<Navbar
  className="navbar-expand"
  color="faded"
  expand={false}
  id="more-recipes-navbar"
  light={true}
  tag="nav"
>
  <Link
    className="navbar-brand"
    replace={false}
    to="/"
  >
    MORE RECIPES
  </Link>
  <div
    className="collapse navbar-collapse justify-content-between"
    id="navbarNav"
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <UncontrolledDropdown
        className="justify-content-start notifications-link"
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          color="secondary"
          nav={true}
        >
          <FontAwesome
            name="bell"
            size="lg"
          />
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <div
            className="container-fluid"
          >
            <p>
              You have a new review
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              3 people viewed your recipe
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              Review your security details
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              Chef Steve just posted a new recipe
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              1 person added your recipe to their favorites
            </p>
          </div>
        </DropdownMenu>
      </UncontrolledDropdown>
      <UncontrolledDropdown
        nav={true}
      >
        <div
          className="nav-profile-picture-div rounded-cirle menu-dropdown d-inline"
        >
          <DropdownToggle
            aria-haspopup={true}
            color="secondary"
            nav={true}
          >
            <img
              alt="User"
              className="rounded-cirle"
              src="images/user1.jpg"
            />
             
            Dave
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <Link
              className="dropdown-item"
              replace={false}
              to="/user/profile"
            >
              Profile
            </Link>
            <Link
              className="dropdown-item"
              replace={false}
              to="/user/profile/edit"
            >
              Edit Profile
            </Link>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/login"
              onClick={[Function]}
            >
              Log out
            </a>
          </DropdownMenu>
        </div>
      </UncontrolledDropdown>
      <UncontrolledDropdown
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          className="menu-dropdown"
          color="secondary"
          nav={true}
        >
          <FontAwesome
            name="reorder"
            size="lg"
          />
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <a
            className="dropdown-item add-recipe"
            data-toggle="modal"
            href="#add-edit-modal"
            title="New Recipe"
          >
            Add a Recipe
          </a>
          <Link
            className="dropdown-item"
            replace={false}
            to="/recipes"
          >
            All Recipes
          </Link>
          <Link
            className="dropdown-item"
            replace={false}
            to="/"
          >
            Top Recipes
          </Link>
        </DropdownMenu>
      </UncontrolledDropdown>
    </Nav>
  </div>
</Navbar>
`;
